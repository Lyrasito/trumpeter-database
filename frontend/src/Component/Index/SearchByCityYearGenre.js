import React from "react";
import { ReactComponent as SearchLogo } from "../../svg/Search.svg";

class SearchByCityYearGenre extends React.Component {
  state = {
    city: "",
    year: "",
    genre: "",
  };

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  };

  search = () => {
    this.props.searchPlayers(
      this.state.city,
      this.state.year,
      this.state.genre
    );
    this.setState({ city: "", year: "", genre: "" });
    this.props.haveSearched();
  };

  render() {
    return (
      <div>
        <div className="searchbar">
          <label>
            City
            <input
              className="search"
              id="citySearch"
              name="city"
              onChange={this.handleChange}
              value={this.state.city}
            ></input>
          </label>
          <label>
            Year
            <input
              className="search"
              id="yearSearch"
              name="year"
              onChange={this.handleChange}
              value={this.state.year}
            ></input>
          </label>
          <label>
            Genre
            <input
              className="search"
              id="genreSearch"
              name="genre"
              onChange={this.handleChange}
              value={this.state.genre}
            ></input>
          </label>
        </div>
        <button className="submit" onClick={this.search}>
          Search
          <SearchLogo />
        </button>
      </div>
    );
  }
}

export default SearchByCityYearGenre;
