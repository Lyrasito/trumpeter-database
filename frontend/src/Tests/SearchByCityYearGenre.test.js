import SearchByCityYearGenre from "../Component/Index/SearchByCityYearGenre";
import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

describe("<SearchByCityYearGenre />", () => {
  it("renders properly", () => {
    const wrapper = shallow(<SearchByCityYearGenre />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
