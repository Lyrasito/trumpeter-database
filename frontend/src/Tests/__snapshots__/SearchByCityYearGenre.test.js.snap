// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchByCityYearGenre /> renders properly 1`] = `
<div>
  <div
    className="searchbar"
  >
    <label>
      City
      <input
        className="search"
        id="citySearch"
        name="city"
        onChange={[Function]}
        value=""
      />
    </label>
    <label>
      Year
      <input
        className="search"
        id="yearSearch"
        name="year"
        onChange={[Function]}
        value=""
      />
    </label>
    <label>
      Genre
      <input
        className="search"
        id="genreSearch"
        name="genre"
        onChange={[Function]}
        value=""
      />
    </label>
  </div>
  <button
    className="submit"
    onClick={[Function]}
  >
    Search
    <ForwardRef(SvgSearch) />
  </button>
</div>
`;
